env.example

```diff
@@ -57,7 +57,12 @@ BLOG_INACTIVE_TIME=360
 GITHUB_TOKEN=
 
 # Feed job queue attempts
-FEED_QUEUE_ATTEMPTS=8
+FEED_QUEUE_ATTEMPTS=2
```

Comment:
Although I agree 8 is a bit high, should this have a default variable of anything higher then 1?

If the individual isn't using the feed queue at the time, shouldn't this just be 1 to maximize efficiency for them? 

I get it from a more post-production standpoint we would play with this number more, but for development purposes wouldn't it be best for the developer to choose depending on what they need to test.

Reply:
We can play with these numbers and try to find a balance.  As you point out, any dev, and any server deployment, can override them with whatever value they like.  So a dev doing testing, who doesn't care about retrying feeds, can set it to 1 (or 0).  Whereas, in production, we can be more liberal with what we retry.

The way that these numbers work in practice is that when a job fails (e.g., can't download a feed, feed can't be parsed, etc.) it will get put into a "delayed" state, and Bull will automatically retry it later.

The number of times it will be retried is `FEED_QUEUE_ATTEMPTS` and `FEED_QUEUE_DELAY_MS` is how long to delay each retry.  NOTE: we are using an `exponential` backoff strategy, which means it will wait 30s (30,000ms) before retrying the first time, 60s the second, and so on (attempts * delay) until it finally works or gives up.

From experience running with the changes in this PR for a while, if I have too many attempts, and wait too long between attempts, the delayed jobs start to back up.

There are a few long-term fixes we can do for this, including better ways to deal with errors (see #548), blacklisting blogs that fail and won't ever work, and also playing with these numbers.

Does this make sense?

Reply:
Yes, thanks!